Dockerfile

```diff
@@ -370,40 +407,53 @@ RUN if [ "${CPAN_PKGS}" != "" ] || [ "${PIP_PKGS}" != "" ] || [ "${IMAGE_LAYER_P
       && rm -rf /var/lib/apt/lists/* /tmp/* /var/tmp/* ~/.[^.] ~/.??* ~/* \
     ; fi
 
+# Custom image layers options:
+ARG IMAGE_LAYER_PYTHON="1"
+ARG IMAGE_LAYER_PYTHON_EXT="1"
+
+# Custom installation packages
+ARG PIP_PKGS=""
+
 # Add Python app layer
 RUN if [ "${PIP_PKGS}" != "" ] || [ "${IMAGE_LAYER_PYTHON}" != "0" ] || [ "${IMAGE_LAYER_PYTHON_EXT}" != "0" ]; then \
       LC_ALL=C DEBIAN_FRONTEND=noninteractive apt-get update \
       && LC_ALL=C DEBIAN_FRONTEND=noninteractive apt-get install -qqy --no-install-recommends \
-          libinline-python-perl \
-          python3 \
-          python3-dev \
-          python3-pip \
-          python3-setuptools \
-          python3-wheel \
+        python3=3.7.3-1 \
+        python3-dev=3.7.3-1 \
+        python3-pip=18.1-5 \
+        python3-setuptools=40.8.0-1 \
+        python3-wheel=0.32.3-2 \
       && if [ "${PIP_PKGS}" != "" ]; then \
            pip3 install \
             ${PIP_PKGS} \
          ; fi \
       && if [ "${IMAGE_LAYER_PYTHON_EXT}" != "0" ]; then \
            LC_ALL=C DEBIAN_FRONTEND=noninteractive apt-get install -qqy --no-install-recommends \
-            python3-pychromecast \
-            speedtest-cli \
-            youtube-dl \
-           && ln -s ../../bin/speedtest-cli /usr/local/bin/speedtest-cli \
```

Comment:
pinning was intentionally left out because there is no real value not to use the latest and greatest version...

Reply:
pinning is implemented, to invalidate the cache if a new version should be uses

Reply:
Here a reference what can bei the result If not aware of a compatible Version:

https://forum.fhem.de/index.php/topic,119201.0.html